rust_src/src/keyboard.rs

```diff
@@ -109,25 +109,25 @@ pub fn posn_at_x_y(
 /// Return true if EVENT is a list whose elements are all integers or symbols.
 /// Such a list is not valid as an event,
 /// but it can be a Lucid-style event type list.
-pub fn lucid_event_type_list_p(event: LispObject) -> bool {
-    if !event.is_cons() {
-        return false;
-    }
-
-    let first = event.as_cons_or_error().car();
-    if first.eq(LispObject::from(Qhelp_echo)) || first.eq(LispObject::from(Qvertical_line))
-        || first.eq(LispObject::from(Qmode_line)) || first.eq(LispObject::from(Qheader_line))
-    {
-        return false;
-    }
+pub fn lucid_event_type_list_p(event: Option<LispCons>) -> bool {
+    event.map_or(false, |event| {
+        let first = event.car();
+        if first.eq(LispObject::from_raw(Qhelp_echo))
+            || first.eq(LispObject::from_raw(Qvertical_line))
+            || first.eq(LispObject::from_raw(Qmode_line))
+            || first.eq(LispObject::from_raw(Qheader_line))
+        {
+            return false;
+        }
 
-    let mut it = event.iter_cars_safe();
+        let mut it = event.as_obj().iter_cars_safe();
```

Comment:
Here is that inconsistency we talked about. This one is `as_obj` elsewhere it is `as_lisp_obj`